tutorials/networking/http_request_class.rst

```diff
@@ -74,58 +95,60 @@ Below is all the code we need to make it work. The URL points to an online API m
+Note that you may want to check whether the ``result`` equals ``RESULT_SUCCESS``
+and whether a JSON parsing error occurred, see the JSON class reference and
+:ref:`HTTPRequest <class_HTTPRequest>` for more.
 
-Of course, you can also set custom HTTP headers. These are given as a string array, with each string containing a header in the format ``"header: value"``.
-For example, to set a custom user agent (the HTTP ``user-agent`` header) you could use the following:
+You have to wait for a request to finish before sending another one.
+Making multiple request at once requires you to have one node per request.
+A common strategy is to create and delete HTTPRequest nodes at runtime as necessary.
+
+Sending data to the server
+--------------------------
+
+Until now, we have limited ourselves to requesting data from a server.
+But what if you need to send data to the server? Here is a common way of doing it:
 
 .. tabs::
 
     .. code-tab:: gdscript GDScript
 
-        $HTTPRequest.request("http://www.mocky.io/v2/5185415ba171ea3a00704eed", ["user-agent: YourCustomUserAgent"])
+        var json = JSON.stringify(data_to_send)
+        var headers = ["Content-Type: application/json"]
+        $HTTPRequest.request(url, headers, HTTPClient.METHOD_POST, json)
 
     .. code-tab:: csharp
 
+        string json = Json.Stringify(dataToSend);
+        string[] headers = new string[] { "Content-Type: application/json" };
         HttpRequest httpRequest = GetNode<HttpRequest>("HTTPRequest");
-        httpRequest.Request("http://www.mocky.io/v2/5185415ba171ea3a00704eed", new string[] { "user-agent: YourCustomUserAgent" });
-
-Please note that, for SSL/TLS encryption and thus HTTPS URLs to work, you may need to take some steps as described :ref:`here <doc_ssl_certificates>`.
+        httpRequest.Request(url, headers, HttpClient.Method.Post, json);
 
-Also, when calling APIs using authorization, be aware that someone might analyse and decompile your released application and thus may gain access to any embedded authorization information like tokens, usernames or passwords.
-That means it is usually not a good idea to embed things such as database access credentials inside your game. Avoid providing information useful to an attacker whenever possible.
-
-Sending data to server
-----------------------
+Setting custom HTTP headers
+---------------------------
 
-Until now, we have limited ourselves to requesting data from a server. But what if you need to send data to the server? Here is a common way of doing it:
+Of course, you can also set custom HTTP headers. These are given as a string array,
+with each string containing a header in the format ``"header: value"``.
+For example, to set a custom user agent (the HTTP ``user-agent`` header) you could use the following:
```

Comment:
I think by convention headers are supposed to be capitalized, and you capitalized `Content-Type` above. So for consistency I'd do it here (and the example below) as well. Can also do it to the format, e.g. `Header: value`.

Reply:
Ah, this was just how it was before my changes.
While the RFC spec defines them as case insensitive, I agree :) Its common convention, and we indeed use that elsewhere.